src/modules/layout/containers/HeaderContainer.js

```diff
@@ -2,6 +2,23 @@
 
 import Header from '../components/Header'
 import { withTheme } from 'styled-components/native'
+import type { StateType } from '../../app/StateType'
+import { connect } from 'react-redux'
+import React from 'react'
+import { withNavigation } from 'react-navigation'
 
-// $FlowFixMe
-export default withTheme(Header)
+const mapStateToProps = (state: StateType, ownProps) => {
+  const key: string = ownProps.navigation.getParam('key')
+  // todo add availableLanguages for other routes
+  const targetRoute = state.cityContent.categoriesRouteMapping[key]
+  const model = targetRoute && targetRoute.root && targetRoute.models[targetRoute.root]
+  const cityLanguages = state.cityContent.languages && state.cityContent.languages.map(languageModel => languageModel.code)
+  const availableLanguages = model && model.title !== state.cityContent.city ? Array.from(model.availableLanguages.keys()) : cityLanguages
```

Comment:
I think we should create a helper for stuff getting stuff like this. We have the reselect library which allows you to write selectors: https://github.com/reduxjs/reselect

This is not needed so far but I think it would make sense to write a function to query the store for values by route key.

Reply:
Still, I think using `model.title` to check if `model` represents the root (at least that's what I think you want to do) is not good. Can't you use `model.isRoot()` instead?

Reply:
@maxammann using the reselect library now to select the availableLanguages. Could you please take a look if this is allright and if the location in `modules/common/selectors/` is okay?

Reply:
Looks fine :+1: